src/Workspaces/CoreTest/Renamer/RenamerTests.CSharp.cs

```diff
@@ -0,0 +1,240 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Microsoft.CodeAnalysis.UnitTests.Renamer
+{
+    public partial class RenamerTests
+    {
+        [Fact]
+        public Task CSharp_TestEmptyDocument()
+        => TestRenameDocumentCSharp(
+            "",
+            "",
+            newDocumentName: "NewDocumentName");
+
+        [Fact]
+        public Task CSharp_RenameDocument_NoRenameType()
+        => TestRenameDocumentCSharp(
+            @"class C {}",
+            @"class C {}",
+            expectRename: false);
+
+        [Fact]
+        public Task CSharp_RenameDocument_RenameType()
+        => TestRenameDocumentCSharp(
+            $@"class {DefaultDocumentName} {{}}",
+            @"class NewDocumentName {}",
+            newDocumentName: "NewDocumentName");
+
+        [Fact]
+        public Task CSharp_RenameDocument_RenameInterface()
+        => TestRenameDocumentCSharp(
+            @"interface IInterface {}",
+            @"interface IInterface2 {}",
+            documentName: "IInterface.cs",
```

Comment:
Should all the tests just pass documentName? The DefaultDocumentName doesn't seem to carry the complexity weight.

Reply:
Sure